src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/RecoveryServicesCmdletBase.cs

```diff
@@ -133,6 +135,117 @@ public void HandleException(Exception ex)
+                    this.recoveryServicesClient.ClientRequestId,
+                    Environment.NewLine);
+            }
+
+            CloudException cloudException = ex as CloudException;
+            if (cloudException != null)
+            {
+                DataContractSerializer deserializer = null;
+
+                try
+                {
+                    using (Stream stream = new MemoryStream())
+                    {
+                        if (cloudException.Error != null && cloudException.Error.OriginalMessage != null)
+                        {
+                            string message = cloudException.Error.OriginalMessage;
+                            byte[] data = System.Text.Encoding.UTF8.GetBytes(message);
+                            stream.Write(data, 0, data.Length);
+                            stream.Position = 0;
+
+                            if (message.Contains("http://schemas.microsoft.com/wars"))
+                            {
+                                deserializer = new DataContractSerializer(typeof(StartVaultUpgradeError));
+                                StartVaultUpgradeError error = (StartVaultUpgradeError)deserializer.ReadObject(stream);
+                                string msg = string.Format(
+                                    Properties.Resources.StartVaultUpgradeExceptionDetails,
+                                    clientRequestIdMsg,
+                                    error.Message.Value);
+                                this.ThrowTerminatingError(
+                                    new ErrorRecord(
+                                        new InvalidOperationException(msg),
+                                        string.Empty,
+                                        ErrorCategory.InvalidOperation,
+                                        null));
+                            }
+                            else
+                            {
+                                deserializer = new DataContractSerializer(typeof(VaultUpgradeError));
+                                VaultUpgradeError error = (VaultUpgradeError)deserializer.ReadObject(stream);
+                                if (error.Details != null)
+                                {
+                                    return this.recoveryServicesClient.ParseErrorDetails(error.Details, clientRequestIdMsg);
+                                }
+                                else
+                                {
+                                    StringBuilder exceptionMessage = new StringBuilder();
+                                    exceptionMessage.AppendLine(Properties.Resources.VaultUpgradeExceptionDetails).AppendLine(" ");
+                                    exceptionMessage.Append(this.recoveryServicesClient.ParseError(error));
+                                    exceptionMessage.AppendLine(clientRequestIdMsg);
+                                    this.ThrowTerminatingError(
```

Comment:
@neetus15 I had forgotten to click on submit. But this can be ignore as just add it to the list of things that needs to be done after the merge into a separate task.